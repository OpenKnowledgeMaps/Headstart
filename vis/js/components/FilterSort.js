import React from "react";
import { connect } from "react-redux";

import StandardFilterSort from "../templates/filtersort/StandardFilterSort";
import BasicFilterSort from "../templates/filtersort/BasicFilterSort";
import ButtonFilterSort from "../templates/filtersort/ButtonFilterSort";

const FilterSort = ({ showList, showFilter, showDropdownSort, color }) => {
  if (showFilter) {
    return <StandardFilterSort displaySort={showList} color={color} />;
  }

  if (showDropdownSort) {
    return <BasicFilterSort displaySort={showList} color={color} />;
  }

  return <ButtonFilterSort displaySort={showList} />;
};

const mapStateToProps = (state) => ({
  showList: state.list.show,
  showFilter: state.list.showFilter,
  showDropdownSort: state.list.showDropdownSort,
  color: state.selectedBubble ? state.selectedBubble.color : null,
});

export default connect(mapStateToProps)(FilterSort);
