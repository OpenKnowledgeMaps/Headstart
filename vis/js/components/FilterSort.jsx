import React from "react";
import { connect } from "react-redux";

import StandardFilterSort from "../templates/filtersort/StandardFilterSort";
import BasicFilterSort from "../templates/filtersort/BasicFilterSort";

const FilterSort = ({ showList, showFilter, color }) => {
  if (showFilter) {
    return <StandardFilterSort displaySort={showList} color={color} />;
  }

  return <BasicFilterSort displaySort={showList} color={color} />;
};

const mapStateToProps = (state) => ({
  showList: state.list.show,
  showFilter: state.list.showFilter,
  color: state.selectedBubble ? state.selectedBubble.color : null,
});

export default connect(mapStateToProps)(FilterSort);
