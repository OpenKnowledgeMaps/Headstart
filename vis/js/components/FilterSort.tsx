// @ts-nocheck
import React from "react";
import { connect } from "react-redux";

import StandardFilterSort from "../templates/filtersort/StandardFilterSort";
import BasicFilterSort from "../templates/filtersort/BasicFilterSort";

export interface FilterSortProps {
  showList: boolean;
  showFilter: boolean;
  color: string;
}

const FilterSort = ({ showList, showFilter, color }: FilterSortProps) => {
  if (showFilter) {
    return <StandardFilterSort displaySort={showList} color={color} />;
  }

  return <BasicFilterSort displaySort={showList} color={color} />;
};

const mapStateToProps = (state) => ({
  showList: state.list.show,
  showFilter: state.list.showFilter,
  color: state.selectedBubble ? state.selectedBubble.color : null,
});

export default connect(mapStateToProps)(FilterSort);
