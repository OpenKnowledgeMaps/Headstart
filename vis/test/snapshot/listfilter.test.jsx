import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";

import FilterSort from "../../js/components/FilterSort";

const mockStore = configureStore([]);

describe("List filter component snapshot", () => {
  it("matches a snapshot", () => {
    const store = mockStore({
      list: {
        show: true,
        searchValue: "case study",
        showFilter: true,
        filterValue: "all",
        filterOptions: ["all", "open_access"],
        sortValue: "relevance",
        sortOptions: ["relevance", "year"],
      },
      localization: {
        search_placeholder: "Search...",
        all: "All",
        open_access: "Open access",
        filter_by_label: "Filter by: ",
        relevance: "Relevance",
        year: "Year",
        sort_by_label: "Sort by: ",
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <FilterSort />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
