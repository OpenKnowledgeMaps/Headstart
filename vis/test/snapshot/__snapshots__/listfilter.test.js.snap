// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List filter component snapshot matches a snapshot 1`] = `
<div
  className="row"
  id="explorer_options"
  style={
    Object {
      "borderBottom": "",
    }
  }
>
  <div
    id="filter_container"
  >
    <div
      className="input-group input-group-sm"
    >
      <i
        className="fa fa-search searchfield-icon"
      />
      <input
        className="form-control"
        id="filter_input"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value="case study"
      />
      <span
        className="glyphicon glyphicon-remove-circle"
        id="searchclear"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="dropdown"
    id="filter_parameter_container"
  >
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="filter_params"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Filter by: 
         
        <span
          id="curr-filter-type"
        >
          All
        </span>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="filter_params"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className="active"
          role="presentation"
        >
          <a
            href="#"
            id="filter_option_all"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            All
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="filter_option_open_access"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            Open access
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="dropdown"
    id="sort_container"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="sort"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Sort by: 
         
        <span
          id="curr-sort-type"
        >
          Relevance
        </span>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="sort"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className="active"
          role="presentation"
        >
          <a
            href="#"
            id="sort_option_relevance"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            Relevance
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="sort_option_year"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            Year
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
